modules/vaos/app/services/eps/provider_service.rb

```diff
@@ -141,82 +154,57 @@ def address_matches?(provider, address)
 
       provider_address = provider[:location][:address]
 
-      # Step 1: Primary filter - check if street address matches (most likely to filter out non-matches)
-      return false unless street_address_matches?(provider_address, address[:street1])
-
-      # Step 2: Full address comparison with logging for monitoring, so we can detect formatting differences
-      full_match = full_address_matches?(provider_address, address)
+      # Compare the two reliable components: street and 5-digit zip
+      street_matches = street_address_matches?(provider_address, address[:street1])
+      zip_matches = zip_code_matches?(provider_address, address[:zip])
 
-      log_partial_address_match(provider_address, address) unless full_match
+      # Log for monitoring if some components match but not all (helps identify format issues)
+      if zip_matches && !street_matches
+        Rails.logger.warn("Provider address partial match - Street: #{street_matches}, " \
```

Comment:
Note: This is also for debugging purposes. No PII here since this is a provider's address.